processing/src/main/java/io/druid/segment/data/VSizeLongSerde.java

```diff
@@ -0,0 +1,634 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.segment.data;
+
+import com.metamx.common.IAE;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.BufferOverflowException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+/**
+ * Currently only support big endian
+ * <p>
+ * An empty 4 bytes is written upon closing to avoid index out of bound exception for deserializers that shift bytes
+ */
+public class VSizeLongSerde
+{
+
+  public static final int SUPPORTED_SIZE[] = {1, 2, 4, 8, 12, 16, 20, 24, 32, 40, 48, 56, 64};
+  public static final byte EMPTY[] = {0, 0, 0, 0};
+
+  public static int getBitsForMax(long value)
+  {
+    if (value < 0) {
+      throw new IAE("maxValue[%s] must be positive", value);
+    }
+    byte numBits = 0;
+    long maxValue = 1;
+    for (int i = 0; i < SUPPORTED_SIZE.length; i++) {
+      while (numBits < SUPPORTED_SIZE[i] && maxValue < Long.MAX_VALUE / 2) {
+        numBits++;
+        maxValue *= 2;
+      }
+      if (value <= maxValue || maxValue >= Long.MAX_VALUE / 2) {
+        return SUPPORTED_SIZE[i];
+      }
+    }
+    return 64;
+  }
+
+  public static int getSerializedSize(int longSize, int numValues)
+  {
+    return (longSize * numValues + 7) / 8 + 4;
```

Comment:
can you add a comment indicating how the serialized size is calculated? what do the 7 and +4 represent?